### PR TITLE
fix(scheduling): replace exact-match open-shift count with coverage overlap

### PR DESCRIPTION
**Pull request by jdelgado2002**
## Summary

- **Root cause fixed:** The "needs staff" banner and `get_open_shifts` RPC were using exact start+end+position matching to decide if a slot was covered. Mid-shift fill-ins and custom-time shifts never matched, causing covered slots to show as needing staff (confirmed on prod: 50/109 shifts matched no template, 24 phantom open slots).
- **Coverage engine added** (`src/lib/shiftCoverage.ts`): a pure TypeScript sweep-line algorithm computes per-minute concurrent distinct-employee headcount, handles overnight windows and restaurant IANA timezone, and produces `minConcurrent`, `openSpots`, `coveragePct`, `segments`, and `coveringEmployees`.
- **SQL mirror** (`shift_slot_min_concurrent`): identical sweep logic in SQL, `STABLE SECURITY DEFINER`, used by both `get_open_shifts` and `claim_open_shift` so the offer and claim guard always agree — no double-claim window.
- **Scheduling banner** (`Scheduling.tsx`): `openShiftCount` now calls the coverage engine instead of `buildTemplateGridData` + `computeOpenSpots` (exact-match path removed).
- **Planner grid** (`ShiftPlannerTab` → `ShiftCell` → `CoverageDetail`): tab-level `useMemo` coverage map; each cell shows an accessible compact indicator button (semantic tokens, `aria-label`, non-color cue, `sr-only`); one lifted `CoverageDetail` popover/Drawer (desktop/mobile) showing covering employees + gap segments — Single Dialog Pattern.
- **Dead code removed:** `computeOpenSpots` and `classifyCapacity` deleted after all callers migrated; raw-color fallback badges replaced with semantic tokens.
- **Performance:** composite index `idx_shifts_restaurant_start_status` in its own `CONCURRENTLY` migration so the coverage sweep stays sub-millisecond per slot.
- **Security:** `shift_slot_min_concurrent` GRANT revoked from `authenticated` directly; callers (`get_open_shifts`, `claim_open_shift`) are `SECURITY DEFINER` and carry the privilege — no cross-tenant headcount disclosure.

Design doc: `docs/superpowers/specs/2026-06-25-open-shift-coverage-design.md`

## Test plan

- [ ] `npm run typecheck` passes with 0 errors
- [ ] `npx vitest run` — all 4717+ unit tests pass, including:
  - `tests/unit/shiftCoverage.test.ts` (15 tests, TZ-portable under UTC/LA/Tokyo)
  - `tests/unit/schedulingOpenShiftCount.test.ts` (7 tests — fill-in covers slot → 0 open)
  - `tests/unit/shiftCellCoverageIndicator.test.tsx` (18 tests — a11y, semantic tokens, memo)
  - `tests/unit/CoverageDetail.test.tsx` (16 tests — popover/Drawer body)
  - `tests/unit/shiftPlannerCoverageWiring.test.ts` (7 source-text assertions)
  - `tests/unit/openShiftHelpersCleanup.test.ts` (3 regression guards)
- [ ] `npm run test:db` — pgTAP green, including:
  - `supabase/tests/open_shift_coverage.test.sql` (5 assertions): fill-in covers full window → `minc=1`; empty → 0; position mismatch → 0; trailing gap → 0; `claim_open_shift` rejects on coverage-full slot (non-tautological fixture)
  - `supabase/tests/open_shifts_capacity_one.test.sql` updated to coverage semantics (4 assertions)
- [ ] Confirm scheduling banner count drops to 0 for Wetzel's–Cold Stone (rid `7c0c76e3`) on week 2026-06-22 where 8/8/5 people were verified working
- [ ] Confirm `get_open_shifts` returns 0 rows for fully-covered weeks
- [ ] `claim_open_shift` on a coverage-full slot returns `{ success: false }` instead of inserting

## Summary by CodeRabbit

* **New Features**
  * Open shift availability now uses true time coverage, so partially overlapping shifts count correctly even when times don’t exactly match.
  * Added a new coverage view in the scheduler showing who is covering a slot, coverage percentage, and any uncovered gaps.
  * Improved mobile and desktop interactions with a single, accessible details panel for coverage.

* **Bug Fixes**
  * Fixed open-shift counts and claim checks to prevent incorrect double-claiming or missing availability.

### CODE REVIEW

**Comment by jdelgado2002** [comment]
**Triage reply — Codex comment on `shift_slot_min_concurrent` (line 71, after-midnight starts):** The migration already documents this v1 limitation at line 71: *"shifts whose local start is p_date-1 but bleed into p_date's early window are not counted. No such data today."* Addressing it would require extending the date filter, adding overnight-split test fixtures to both the SQL pgTAP suite and the TypeScript unit suite, and verifying existing data — out of scope for this PR. A follow-up issue will track the after-midnight carry-over edge case. Declining in this PR.

File: src/components/scheduling/ShiftPlanner/ShiftCell.tsx [hunk 1]
```diff
@@ -61,6 +87,10 @@ export const ShiftCell = memo(
       allocationStatus === 'available' && 'bg-primary/5',
     );
 
+    // Suppress coverage indicator when fully covered with 0 or 1 placed shifts (noise reduction).
+    const showCoverageIndicator = coverage !== undefined &&
+      !(coverage.coveragePct === 100 && shifts.length <= 1);
+
```
[the remark this answers is not in the document]
Comment (jdelgado2002) [reply src/components/scheduling/ShiftPlanner/ShiftCell.tsx]: Addressed in d216d74b — the core bug (hiding a covered slot when `shifts.length === 0`) is fixed: the guard is now `!(coveragePct === 100 && shifts.length >= 1)`, so a slot covered only by a non-template fill-in (0 bucketed shifts) still renders the indicator. We deliberately kept the `shifts.length >= 1` form rather than adding `coveringEmployees.length <= 1`: when an assignee chip is already rendered in the cell, the 100%-covered coverage chip is redundant noise — its value is the *gap* signal on under-covered slots. Declining the stylistic variant; the correctness concern is resolved.
